syft/cataloger/javascript/cataloger.go

```diff
@@ -8,11 +8,20 @@ import (
 )
 
 // NewJavascriptCataloger returns a new JavaScript cataloger object.
-func NewJavascriptCataloger() *common.GenericCataloger {
+func NewJavascriptPackageCataloger() *common.GenericCataloger {
 	globParsers := map[string]common.ParserFn{
+		"**/package.json": parsePackageJSON,
+	}
+
+	return common.NewGenericCataloger(nil, globParsers, "javascript-package-cataloger")
+}
+
+// NewJavascriptCataloger returns a new Javascript cataloger based on package.json.
+func NewJavascriptLockCataloger() *common.GenericCataloger {
+	globalParsers := map[string]common.ParserFn{
 		"**/package-lock.json": parsePackageLock,
 		"**/yarn.lock":         parseYarnLock,
 	}
 
-	return common.NewGenericCataloger(nil, globParsers, "javascript-cataloger")
+	return common.NewGenericCataloger(nil, globalParsers, "javascript-lock-cataloger")
```

Comment:
I am not entirely sure that `globalParsers` is accurate here, did you mean to use `globParsers` instead? This is coming from the usage of globs...